robot/Cumulus/resources/OpportunityPageObject.py

```diff
@@ -14,6 +14,13 @@ def _is_current_page(self):
         """
         self.selenium.wait_until_location_contains("/lightning/r/Opportunity/",message="Current page is not a Opportunity detail view")
 
+    def ensure_opportunity_details_are_loaded(self,objectID, value):
+        """ Navigate to the page with objectid mentioned
+           Wait for the page to load and confirm atleast the opportunity name exists
+        """
+        self.pageobjects.go_to_page("Details", "Opportunity", objectID)
+        self.npsp.navigate_to_and_validate_field_value("Opportunity Name", "contains", value)
```

Comment:
you can directly call wait _until_page_contains_element (locator,timeout=60) instead of this keyword and avoid wait until keyword succeeds in robot file

Reply:
u can name it as go to page and ensure opportunity details are loaded

Reply:
The reason I added this keyword, is this keyword does polling, When the page does not load data for any unknown environmental reason , it will try the same command in that specified duration. I researched about this keyword on google forums. They say it helps in situations like these. In this case, first time when the page is loaded incompletely and it could not find the specified element, it will execute the same keyword, so tries to reload the page and tries to find the element. repeats the same steps until the keyword succeeds in that span of time specified.

Reply:
Wait Until Keyword Succeeds | retry, retry_interval, name, *args | Runs the specified keyword and retries if it fails.name and args define the keyword that is executed similarly as with Run Keyword. How long to retry running the keyword is defined using retry argument either as timeout or count. retry_interval is the time to wait before trying to run the keyword again after the previous run has failed.
-- | -- | --